src/openfluid/ui/waresdev/WareStatusDashboardWidget.cpp

```diff
@@ -0,0 +1,231 @@
+// =====================================================================
+
+
+WareStatusDashboardWidget::~WareStatusDashboardWidget()
+{
+  delete ui;
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareStatusDashboardWidget::addMessage(openfluid::waresdev::WareSrcChecker::ReportingData::ReportingItem Item)
+{
+  WareStatusItemWidget* MessageWidget = new WareStatusItemWidget(Item);
+  mp_MessageContainer->layout()->addWidget(MessageWidget);
+
+  connect(MessageWidget, SIGNAL(actionTriggered(QString)), this, SLOT(onActionTriggered(QString))); 
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareStatusDashboardWidget::clear()
+{
+  setVisible(false);
+  while (QLayoutItem* item = mp_MessageContainer->layout()->takeAt(0))
+  {
+    if (item->widget())
+    {
+      delete item->widget();
+    }
+    delete item;
+  }
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareStatusDashboardWidget::refresh()
+{
+  // First clean existing data
+  clear();
+  
+  try // TOIMPL improve try/catch filtering
+  {
+    auto Checker = openfluid::waresdev::WareSrcChecker(m_WarePath);
+    auto Report = Checker.performCheck(true);
+    
+    setLevel(Report.getStatus());
+
+    // TOIMPL check order
+    for (const auto& CatItems : Report.Categories)
+    {
+
+      for (const auto& Item : CatItems.second.Items)
+      {
+        if (Item.Status == ReportingData::ReportingStatus::WARNING || 
+            Item.Status == ReportingData::ReportingStatus::ERROR)
+        {
+          addMessage(Item);
+        }
+      }
+    }
+  }
+  catch (std::exception& E)
+  {
+    std::string ErrorMsg = "WareSrcChecker error: " + std::string(E.what());
+    setLevel(ReportingData::ReportingStatus::DISABLED);
+    ui->TitleLabel->setToolTip(QString::fromStdString(ErrorMsg));
+  }
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareStatusDashboardWidget::onActionTriggered(const QString& ActionCode)
+{
+  if (ActionCode == "migration")
+  {
+    emit migrationRequested();
+  }
+  else
+  {
+    throw openfluid::base::FrameworkException(OPENFLUID_CODE_LOCATION, "Board triggered action not recognized");
+  }
+}
+
+
+// =====================================================================
+// =====================================================================
+
+
+void WareStatusDashboardWidget::setLevel(const ReportingData::ReportingStatus Level)
+{
+  if (Level == ReportingData::ReportingStatus::OK)
+  {
+    ui->TitleLabel->setText(tr("No issue"));
+    m_TitleColor = "#55B332";
```

Comment:
Colors values already defined elsewhere (in Builder?).
Should be variablelized?